src/app/content/practiceQuestions/hooks/locationChange.ts

```diff
@@ -22,11 +21,26 @@ const loadSummary = async(services: MiddlewareAPI & AppServices): Promise<Practi
 };
 
 const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
-  const practiceQuestionsSummary = await loadSummary(services);
+  const state = services.getState();
+  const { book, page } = bookAndPage(state);
+  const isEnabled = practiceQuestionsEnabled(state);
 
-  if (!practiceQuestionsSummary) { return; }
+  if (!book || !page || !isEnabled ) { return; }
 
-  services.dispatch(receivePracticeQuestionsSummary(practiceQuestionsSummary));
+  let practiceQuestionsSummary;
+  if (!hasPracticeQuestions(services.getState())) {
```

Comment:
why do you have all this conditional setting of `practiceQuestionsSummary` like this? the same condition noops down on line 38, can't you just move that up?

Reply:
1st `hasPracticeQuestion` could be actually changed into `practiceQuestionsSummary` selector to check if summary isn't already loaded.
2nd `hasPracticeQuestion` checks if questionSummary actually has questions.

When user enters a page with PQ enabled, `loactionChange` hook loads the PQ summary. The first if statement prevents the summary reload after opening the modal.